### DISCUSSION TITLE
MacOS support

### DISCUSSION THREAD

**Discussion by GuardKenzie** [GuardKenzie/chafa.py#7] [General]
I want to get chafa.py to support MacOS since chafa does, however, this is proving to be a bit of a challenge. If anyone is interested in helping out or has any ideas, please reply to this discussion thread. After doing some reading on using ctypes with homebrew and this is the conclusion:

### The mess

After apple introduced Apple silicon, homebrew moved the installation location of libraries from `/usr/local` to `/opt/homebrew` (See [this discussion](https://github.com/Homebrew/brew/issues/9177)). The Python `ctypes` library does not search for `.dylib` files in this path but homebrew seems to ship a patched version of python which does (See [this discussion](https://github.com/Homebrew/discussions/discussions/3424) and this [stack overflow post](https://stackoverflow.com/questions/31148387/which-paths-does-python-ctypes-module-search-for-libraries-on-mac-os))

I can see two ways of moving forwards:
1. Manually search for `libchafa` and `libglib-2.0` in `/opt/homebrew` (I'm leaning towards this solution)
2. Having part of the requirements for MacOS be using  a brew installed Python
3. Packaging prebuilt binaries with the chafa.py wheel (I cannot do this myself since I lack an apple computer for bulding/testing)

**Comment by reagle** [comment]
1. Hackish, but might work.
2. If there was a chafa.py brew, I could see it requiring that dependency, but I'm using pip and pyenv, for example.
3. Again, I don't know much about packages and distribution, but I wonder if there's github triggers for building platform-specific binaries...?

**Reply by GuardKenzie** [reply]
> I wonder if there's github triggers for building platform-specific binaries...?

I think GitHub actions might be able to do that. I tried using an automated build action for windows and Linux and it also seems to support MacOS. I'll just need to test that on a fork or something.

I would need to build both glib2 and chafa in order for it to work though. Shouldn't be too much of an issue (foreshadowing haha)

**Reply by GuardKenzie** [reply]
@reagle You were right! I managed to build chafa.py for MacOS on GitHub actions through the magic of trial and error (and a bit of luch). There are now built wheels for intel MacOS computers available on PyPi.

The reason for only supporting intel macs for the moment can be found in [this issue](https://github.com/actions/runner-images/issues/2187). The GitHub runners don't support M1 macs yet so I have no way of building on one. If anyone is interested in helping out with that, I'd be delighted!

**Comment by hpjansson** [comment]
Hi @fxcoudert, @chenrui333 -- I noticed you've been maintaining Chafa for Homebrew. Thanks a ton for that! Do you know someone who'd be interested in helping @GuardKenzie, or maybe packaging chafa.py? Neither of us has a Mac at the moment, but it'd be cool to have Python bindings there.
